app/services/authentication/authenticator.rb

```diff
@@ -86,6 +86,9 @@ def current_user_identity_exists?
     end
 
     def proper_user(identity)
+      Rails.logger.debug "PROPER USER: #{current_user.inspect}"
+      Rails.logger.debug "IDENTITY: #{identity.inspect}"
+      Rails.logger.debug "--------" * 10
```

Comment:
The bug noticed in https://community.benhalpern.com is that when connecting an existing account with the SIWA OAuth process a new account is created (instead of connecting the user currently logged in).

I'm still debugging this part but I suspect we may be running into a problem related to the session not being available (because the Cookies have `SameSite=Lax`)

Reply:
Hi @fdoxyz, just checking in to see how this debugging is going?

Reply:
Thanks for bringing this up @citizen428!

The conclusion back then was that we need `SameSite=None` in order to make this work and we were blocked by TLS work (now shipped) in order to enforce `Secure` on our cookies. I'll loop back around this throughout the week since it shouldn't 🤞  be too difficult to sort it out now.